server.elm

```diff
@@ -2033,72 +2270,310 @@ editionscript = """
+            }),
+            el("div", {"class": "mainElemInfo"}, textPreview(elem, 50)) // display its text content
+          )
+        }
+
+        let displayChildrenElem = function(elem) {
+          let childrenElemDiv = document.querySelector(".dom-selector > .childrenElem");
+          childrenElemDiv.append(
+            el("div", {"class": "childrenSelector"},
+              [
+                el("div", {"class": "childrenSelectorName"}, "<" + elem.tagName.toLowerCase() + ">", {}),
+                el("div", {"class": "childrenSelectorInfo"}, textPreview(elem, 20))
+              ], 
+              {
+                onmouseenter: (c => () => { c.setAttribute("ghost-hovered", "true") })(elem),
+                onmouseleave: (c => () => { c.removeAttribute("ghost-hovered") })(elem)
+              }
+            )
+          );
+        }
+
+        let displayElemAttr = function(targetDiv, elem) {
+          // show attribute of element on the dom selector
+          for (let i = 0; elem && elem.attributes && i < elem.attributes.length; i++) {
+            let name = elem.attributes[i].name;
+            let value = elem.attributes[i].value;
+            if (name === "ghost-clicked" || name === "ghost-hovered") continue;
+            targetDiv.append(
+              el("div", { "class": "elementAttr" },
+                [
+                  el("span", { title: "This element has attribute name '" + name + "'" }, name + ": "),
+                  el("span", { title: "This element has attribute value '" + value + "'" }, value)
+                ]
+              )
+            );
+          }
+        }
+
+        if (clickedElem.id !== "context-menu" || clickedElem.id !== "modify-menu" || clickedElem.id !== "editbox") {
+          if (selectorStatus === 1) {
+            // status 1
+            // display clicked element in main part
+            let mainElemDiv = document.querySelector(".dom-selector > .mainElem");
+            displayMainElem(clickedElem);
+            domSelectorDiv.classList.add("selectedDom");
+            mainElemDiv.addEventListener('click', (c => event => {
+              // <html> tag cannot be clicked since it has no parent
+              if ((c.tagName && c.tagName === "HTML") || !c.tagName) {
+                return;
+              }
+
+              // switch to status 2
+              selectorStatus = 2;
+              editor_model.clickedElem = c;
+              editor_model.notextselection = true;
+              updateInteractionDiv();
+            })(clickedElem));
+            displayElemAttr(mainElemDiv, clickedElem);
+            // document.querySelector(".mainElem").append(
+            //   el("div", {"class": "selectedElementTag elementTag"}, "selected")
+            // );
+
+            // display children elements in second part
+            // if clickedElement has children elements
+            if (clickedElem.children.length > 0) {
+              // only display first 3 children elements
```

Comment:
I know this is what I asked for, but let me ask you a non trivial task. I fear that if I select an element and want to understand its structure and its ancestors, if I click on the parent, I'll loose track of the previously selected child.
For example, if in statusCode = 2 (clicked element as child), the clicked element is the 23rd child, if I click on its parent once (or three times) and end up in statusCode = 1 with the parent selected, it will always display the first three children, whereas I would be more interested to see the 22nd, 23rd and 24th child as children.
If no child was previously selected, it makes sense to display the first three children.
Would you be able to implement this functionality?
The vision is one of the least surprise. We want to be able to "backtrack", get back to a previous selection state.

For the implementation, I would suggest to store an array in editor_model that is pushed the index of previous child every time a parent is selected. When a child previously selected is selected again, its index is popped. When a new child is selected, the array is erased.

Reply:
OK. I'll try to implement this.